cdap-docs/developers-manual/source/building-blocks/namespaces.rst

```diff
@@ -0,0 +1,67 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _namespaces:
+
+============================================
+Namespaces
+============================================
+
+Overview
+========
+A namespace is a physical grouping of application and data in CDAP. Conceptually,
+namespaces can be thought of as a partitioning of a CDAP instance. Any application or data
+(referred to here as an “entity”) can exist independently in multiple namespaces at the
+same time. The data and metadata of an entity is stored independent of another instance of
+the same entity in a different namespace. 
+
+The primary motivation for namespaces in CDAP is to achieve application and data
+isolation. This is an intial step towards introducing `multi-tenancy
+<http://en.wikipedia.org/wiki/Multitenancy>`__ into CDAP.
```

Comment:
It would great to describe initial use-cases  - There are currently two customer use-cases supporting the work of Namespace
[x] Slicing a single Hadoop Cluster into multiple namespaces for supporting different environments - Dev, QA, Staging
[x] Slicing a single Hadoop Cluster for supporting multiple customers